scalameta/src/main/scala/scala/meta/internal/parsers/Parsers.scala

```diff
@@ -2007,8 +2042,6 @@ private[meta] class Parser(val input: Input)(implicit val dialect: Dialect) { pa
           case p => p.require[Pat]
         }
         makeTuple[Pat](patterns, () => Lit.Unit(), Pat.Tuple(_))
-      case _: Unquote =>
-        unquote[Pat]
```

Comment:
Haha!

```
21:33 ~$ parse 'x match { case foo[bar]() => }'
[[syntax trees at end of parser]]// Scala source: tmpR7L1zs
package <empty> {
  class wrapper extends scala.AnyRef {
    def <init>() = {
      super.<init>();
      ()
    };
    x match {
      case foo[(bar @ <empty>)]() => ()
    }
  }
}
```

This means that Pat.Extract.targs should be `Seq[Pat.Type]`, not `Seq[Type]`.